react-app/src/components/GridPage.tsx

```diff
@@ -0,0 +1,170 @@
+    { id: 5, aspect: "aspect_1_to_3", selected: true },
+  ]);
+
+  const [elements, setElements] = useState(getGeneratedElements);
+
+  const aspectButtons = selectedAspects.map((el, index) => {
+    return (
+      <GridConfigButton
+        key={index}
+        name={el.aspect}
+        handleClick={() => {
+          checkAspect(el.id);
+        }}
+        style={el.selected ? {} : { textDecoration: "line-through" }}
+      />
+    );
+  });
+
+  function getRandomAspect() {
+    const filteredAspects = selectedAspects.filter((e) => {
+      return e.selected == true;
+    });
+
+    if (filteredAspects.length == 0) {
+      return styles[selectedAspects[0].aspect];
+    }
+
+    const randomIndex = Math.floor(Math.random() * filteredAspects.length);
+    return styles[filteredAspects[randomIndex].aspect];
+  }
+
+  function getRandomColor() {
+    const randomIndex = Math.floor(Math.random() * colors.length);
+    return colors[randomIndex];
+  }
+
+  function getGeneratedElements() {
+    const colors: Array<Array<number>> = [];
+    for (let i = 0; i < numberOfElements; i++) {
+      colors.push(getRandomColor());
+    }
+
+    return Array.from({ length: numberOfElements }).map((_, index) => (
+      <div
+        key={index}
+        className={`${styles["element"]} ${getRandomAspect()}`}
+        style={{
+          backgroundColor: "rgb(" + colors[index].join(", ") + ")",
+          borderColor:
+            "rgb(" +
+            subtractWithSaturation8bit(60, colors[index][0]).toString() +
+            ", " +
+            subtractWithSaturation8bit(60, colors[index][1]).toString() +
+            ", " +
+            subtractWithSaturation8bit(60, colors[index][2]).toString() +
+            ")",
+          textAlign: "center",
+          alignContent: "center",
+          fontSize: "2rem",
+          color: "black",
+          overflow: "hidden",
+        }}
+      >
+        <span>{index}</span>
+      </div>
+    ));
+  }
+
+  function setGridCss(newProps: CSSProperties) {
+    const finalProps: CSSProperties = {
+      ...cssProps,
+      ...newProps,
+    };
+
+    setCssProps(finalProps);
+  }
+
+  function setGapSizeHandler(e: React.ChangeEvent<HTMLInputElement>) {
+    setGridCss({
+      gap: e.target.value,
+    });
+  }
+
+  function setElementSizeHandler(e: React.ChangeEvent<HTMLInputElement>) {
+    setGridCss({
+      gridTemplateColumns: e.target.value,
+      gridAutoRows: e.target.value,
+    });
+  }
+
+  function checkAspect(id: number) {
+    const newAspects = [...selectedAspects];
+    newAspects.forEach((element) => {
+      if (element.id == id) {
+        element.selected = !element.selected;
+      }
+    });
+
+    setSelectedAspects(newAspects);
+  }
+
+  return (
+    <div className={styles["page-wrapper"]}>
+      <div className={styles["page-content"]}>
+        <div style={{ width: "100%", backgroundColor: "inherit" }}>
+          <div className={styles["windows-container"]}>
```

Comment:
nesting of divs, also, on my screen, containers are hidden under the navbar
![image](https://github.com/user-attachments/assets/3caeb8d5-b1b1-43ef-b6b3-afb60e331ce1)